build/tools/psd1-to-nuspec.ps1

```diff
@@ -0,0 +1,549 @@
+    ForEach-Object -Process {
+      [System.IO.Path]::GetFileNameWithoutExtension($_)
+    }
+  }
+
+  return $RoleCapabilityNames
+}
+function Join-PathUtility
+{
+  <#
+      .DESCRIPTION
+      Utility to get the case-sensitive path, if exists.
+      Otherwise, returns the output of Join-Path cmdlet.
+      This is required for getting the case-sensitive paths on non-Windows platforms.
+  #>
+  param
+  (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $Path,
+
+    [Parameter(Mandatory = $false)]
+    [string]
+    $ChildPath,
+
+    [Parameter(Mandatory = $false)]
+    [string]
+    [ValidateSet('File', 'Directory', 'Any')]
+    $PathType = 'Any'
+  )
+
+  $JoinedPath = Microsoft.PowerShell.Management\Join-Path -Path $Path -ChildPath $ChildPath
+  if(Microsoft.PowerShell.Management\Test-Path -Path $Path -PathType Container) 
+  {
+    $GetChildItem_params = @{
+      Path          = $Path
+      ErrorAction   = 'SilentlyContinue'
+      WarningAction = 'SilentlyContinue'
+    }
+    if($PathType -eq 'File') 
+    {
+      $GetChildItem_params['File'] = $true
+    }
+    elseif($PathType -eq 'Directory') 
+    {
+      $GetChildItem_params['Directory'] = $true
+    }
+
+    $FoundPath = Microsoft.PowerShell.Management\Get-ChildItem @GetChildItem_params |
+    Where-Object -FilterScript {
+      $_.Name -eq $ChildPath
+    } |
+    ForEach-Object -Process {
+      $_.FullName
+    } |
+    Select-Object -First 1 -ErrorAction SilentlyContinue
+
+    if($FoundPath) 
+    {
+      $JoinedPath = $FoundPath
+    }
+  }
+
+  return $JoinedPath
+}
+function Get-ManifestHashTable
+{
+  param
+  (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string]
+    $Path
+  )
+
+  $Lines = $null
+
+  try
+  {
+    $Lines = Get-Content -Path $Path -Force
+  }
+  catch
+  {
+    Throw "Unable to parse manifest file '$path'"
+    Exit -1
+  }
+
+  if(-not $Lines)
+  {
+    return
+  }
+
+  $scriptBlock = [ScriptBlock]::Create( $Lines -join "`n" )
+
+  $allowedVariables = [System.Collections.Generic.List[String]] @('PSEdition', 'PSScriptRoot')
+  $allowedCommands = [System.Collections.Generic.List[String]] @()
+  $allowEnvironmentVariables = $false
+
+  try
+  {
+    $scriptBlock.CheckRestrictedLanguage($allowedCommands, $allowedVariables, $allowEnvironmentVariables)
+  }
+  catch
+  {
+    return
+  }
+
+  return $scriptBlock.InvokeReturnAsIs()
+}
+
+function New-NuSpecFile
+{
+  [CmdletBinding(PositionalBinding = $false)]
+  Param
+  (
+    [Parameter(Mandatory = $true, ParameterSetName = 'PublishModule')]
+    [ValidateNotNullOrEmpty()]
+    [string]
+    $ManifestPath,
+
+    [Parameter(Mandatory = $false)]
+    [string]
+    $DestinationFolder
+  )
+
+  #variables
+  $Name = $null
+  $Description = $null
+  $Version = ''
+  $Author = $null
+  $CompanyName = $null
+  $Copyright = $null
+  $requireLicenseAcceptance = 'false'
+ 
+  $PSModuleInfo = Test-ModuleManifest -Path $ManifestPath -ErrorAction SilentlyContinue
```

Comment:
Now, I did skip the failures in the `Test-ModuleManifest` though.

Reply:
@tomkerkhove , do you want me to write some extra logic to comment out the `RequiredModules` in the `.psd1` files before we run this and then uncomment them afterwards?
It defeats the purpose a bit by doing so, but then we can fail on other validation failures.

Reply:
If we can and you are up for it, I would give that a try for sure! Better have validation rather than none

Reply:
Ha, I think this is already done here: https://github.com/arcus-azure/arcus.scripting/pull/61/files#diff-75b200261f61d0658f8ca326c14b6373R252

Reply:
😄 , yes, I implemented it just yesterday.